### PR TITLE
Moves editor specific parts of the code to separate AMD modules

### PR DESCRIPTION
**Pull request by juho-jaakkola**
I did this quite quickly, so I'm not completely sure if this is the best way to separate concerns.

Tested with CKEditor and regular textarea. Multiline support for CKEditor is broken, but that was the case also before the PR.

### CODE REVIEW

File: views/default/js/mentions/editors/ckeditor.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/**
+ * CKEditor support for the mentions plugins
+ */
+define(function(require) {
+	var mentions = require('mentions/autocomplete');
+
+	require(['ckeditor'], function(CKEDITOR) {
```
Comment (mrclay) [inline views/default/js/mentions/editors/ckeditor.js]: Why must this be async?

File: views/default/js/mentions/editors/tinymce.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Mentions support for the TinyMCE editor
+ */
+define(function(require) {
+	var mentions = require('mentions/autocomplete');
+
+	// Give some time for the TinyMCE to load
```
Comment (mrclay) [inline views/default/js/mentions/editors/tinymce.js]: :( I guess we don't get any document-level event we could wait for?

File: views/default/js/mentions/editors/ckeditor.js (hunk 1 above)
Comment (juho-jaakkola) [inline views/default/js/mentions/editors/ckeditor.js]: I'm not completely sure in fact. I made multiple quick iterations of this and I'm not sure if it's needed in this one.

File: views/default/js/mentions/editors/tinymce.js (hunk 2 above)
Comment (juho-jaakkola) [inline views/default/js/mentions/editors/tinymce.js]: There might be. AFAIK back when this was made, TinyMCE was not available as AMD module so this was the only way. I'm not sure of the current state.

**Comment by juho-jaakkola** [comment]
What do you think of the overall approach? We can make editor-specific tweaks after merging.

File: views/default/js/mentions/editors/ckeditor.js (hunk 1 above)
Comment (juho-jaakkola) [inline views/default/js/mentions/editors/ckeditor.js]: Yep, not necessary to load async.

**Comment by juho-jaakkola** [comment]
I've now verified that this works also on TinyMCE.

**Comment by mrclay** [comment]
Yeah, approach looks good. :+1:

**Pull request merged**